data-prepper-plugins/dynamodb-source/src/main/java/org/opensearch/dataprepper/plugins/source/dynamodb/stream/ShardAcknowledgementManager.java

```diff
@@ -190,11 +197,13 @@ private void handleFailure(final StreamPartition streamPartition,
         }
         partitionsToRemove.add(streamPartition);
         sourceCoordinator.giveUpPartition(streamPartition);
+        partitionsToGiveUp.remove(streamPartition);
```

Comment:
If it doesn't exist it's a no-op right?

Reply:
Yeah, I had a conditional but IntelliJ warned me it was unnecessary